scripts/mod_loader/modapi/skills.lua

```diff
@@ -104,9 +158,10 @@ SkillEffect.AddGrapple = function(self, source, target, anim, ...)
     metadataTable.source = Point(source.x, source.y)
     metadataTable.target = Point(target.x, target.y)
     metadataTable.anim = anim
-    addDamageListMetadata(self.effect, metadataTable)
 
     oldAddGrapple(self, source, target, anim, ...)
+    -- store metadaata in the grapple space damage
+    addDamageListMetadata(self.effect, metadataTable, true)
```

Comment:
For grapple, we could technically create a dummy SkillEffect instance and call `oldAddGrapple` on that, then inspect the added damage instance to check if it contains a script. Then we can copy it to the actual  SkillEffect.

Though considering that the damage instance is created by the game, and not by lua, I don't think there's a high likelihood of grapple ever having a script attached to it (unless a mod overwrites it after the fact, like we do here), so I'm mentioning this more as food for though.

Reply:
I tested in game and it lacked a script, so I don't expect it to be a problem. It seems grapple is implemented as part of the hidden data not seen in space damage